CONTRIBUTING.md

```diff
@@ -21,25 +21,21 @@ Contributions following these guidelines are always welcomed, encouraged and app
 
 ## Project Release Process
 
-Jazzband aims to give full access to all members, including performing releases, as described in the
-[Jazzband Releases documentation](https://jazzband.co/about/releases).
+Releases require approval by a member of the `pip-tools-leads` team.
 
-To help keeping track of the releases and their changes, here's the current release process:
+This is the current release process:
 
-- Check to see if any recently merged PRs are missing from the milestone of the version about to be released.
-- Create a branch for the release. _Ex: release-3.4.0_.
-- Update the [CHANGELOG] with the version, date and add the text from [drafter release](https://github.com/jazzband/pip-tools/releases).
+- Create a branch for the release. _e.g., `release-3.4.0`_.
+- Use `towncrier` to update the [CHANGELOG], _e.g., `towncrier build --version v3.4.0`_.
 - Push the branch to your fork and create a pull request.
-- Merge the pull request after the changes being approved.
+- Merge the pull request after the changes are approved.
 - Make sure that the tests/CI still pass.
-- Once ready, go to [releases](https://github.com/jazzband/pip-tools/releases) page and publish the latest draft release. This will push a tag on the HEAD of the main branch, trigger the CI pipeline and
-  deploy a pip-tools release in the **Jazzband private package index** upon success.
-- The pip-tools "lead" project members will receive an email notification to review the release and
-  deploy it to the public PyPI if all is correct.
-- Once the release to the public PyPI is confirmed, close the milestone.
-
-Please be mindful of other before and when performing a release, and use this access responsibly.
-
-Do not hesitate to ask questions if you have any before performing a release.
+- Pull the latest changes to `main` locally.
+- Create an unsigned tag with the release version number prefixed with a `v`, _e.g., `git tag v3.4.0`_, and push.
```

Comment:
Better make it annotated
```suggestion
- Create an unsigned tag with the release version number prefixed with a `v`, _e.g., `git tag -a v3.4.0`_, and push.
```

Reply:
Ah, right. Yes. 
Also, might be a good idea to have an explicit reference `git tag -a v3.4.0 -m v3.4.0 upstream/main` (post `git fetch upstream` or `git fetch --all`).

---

Just remembered how we tried to include Git instructions in attrs (https://github.com/python-attrs/attrs/pull/1105 / https://github.com/python-attrs/attrs/pull/1123). But that ended up being removed (https://github.com/python-attrs/attrs/commit/52dfaf00ed6228c1b93e7543db308af70a9b314e).

So maybe this is overkill.